### PR TITLE
feat: AI Command bar

### PR DESCRIPTION
**Pull request by istarkov**
## Description

https://638245578b47d4399fba191a-slxedcgyqd.chromatic.com/?path=/story/library-ai-command-bar--demo

image

image

scroll
image

## Steps for reproduction

Open design-system storybook, find CommandBar

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/design-system/src/components/button.tsx [hunk 1]
```diff
@@ -49,17 +51,25 @@ const foregrounds: Record<ButtonColor, string> = {
   ghost: theme.colors.foregroundMain,
   dark: theme.colors.foregroundContrastMain,
   gradient: theme.colors.foregroundContrastMain,
+  "dark-ghost": theme.colors.foregroundContrastMain,
```
Comment (kof) [inline packages/design-system/src/components/button.tsx:54]: hmm did we use dash before here? or camel case?

Comment (istarkov) [reply packages/design-system/src/components/button.tsx:54]: in figma here is space, but variants are not working with space. For -neutral style above we use dash

File: packages/design-system/src/components/ai-command-bar/command-bar-button.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const RoundedButton = styled(Button, {
+  borderRadius: theme.borderRadius["pill"],
+  width: "min-content",
+});
+
+export const CommandBarButton = forwardRef(
+  ({ children, ...props }: CommandButtonProps, ref: Ref<HTMLButtonElement>) => {
+    return <RoundedButton ref={ref} {...props} prefix={children} />;
+  }
+);
```
Comment (kof) [inline packages/design-system/src/components/ai-command-bar/command-bar-button.tsx:16]: I wonder what this component is for, styled()  is forwarding the ref automatically, doesn't it?

File: packages/design-system/src/components/ai-command-bar/command-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  },
+});
+
+const CollapsibleRoot = styled(Collapsible.Root, {
+  color: theme.colors.foregroundContrastMain,
+  backgroundColor: "#1D1D1D",
+  borderRadius,
+  border: "1px solid #323232",
+  boxShadow:
+    "0px 5px 17px 0px rgba(0, 0, 0, 0.30), 0px 2px 7px 0px rgba(0, 0, 0, 0.10)",
```
Comment (kof) [inline packages/design-system/src/components/ai-command-bar/command-bar.tsx:41]: add a @todo please everywhere where we need to update tokens and work left overs

File: packages/design-system/src/components/ai-command-bar/command-bar-button.tsx (hunk 2 above)
Comment (istarkov) [reply packages/design-system/src/components/ai-command-bar/command-bar-button.tsx:16]: children passed to prefix

File: packages/design-system/src/components/ai-command-bar/command-bar.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      <GridContainer gap={2} open={props.open}>
+        {props.children}
+      </GridContainer>
+    </CollapsibleRoot>
+  );
+};
+
+export const CommandBarTrigger = Collapsible.Trigger;
+
+CommandBarTrigger.defaultProps = {
```
Comment (kof) [inline packages/design-system/src/components/ai-command-bar/command-bar.tsx:66]: defaultProps has been deprecated in react 18

Comment (istarkov) [reply packages/design-system/src/components/ai-command-bar/command-bar.tsx:66]: I know, just toooo much work to overwrite every component to add a new prop. Looking for simple solution

File: packages/design-system/src/components/button.tsx (hunk 1 above)
Comment (kof) [reply packages/design-system/src/components/button.tsx:54]: what have we done so far? did we convert space to something else? or is it first time figma uses space and should be fixed in figma?

Comment (kof) [reply packages/design-system/src/components/button.tsx:54]: @taylornowotny was "dark ghost" first variant that uses space?

File: packages/design-system/src/components/ai-command-bar/command-bar-button.tsx (hunk 2 above)
Comment (kof) [reply packages/design-system/src/components/ai-command-bar/command-bar-button.tsx:16]: you could as well use prefix, seems like a minor detail and not worth a separate component, unless you think there will be much more logic in it

File: packages/design-system/src/components/ai-command-bar/command-bar.tsx (hunk 4 above)
Comment (kof) [reply packages/design-system/src/components/ai-command-bar/command-bar.tsx:66]: yeah, but we can't do that, you are changing the default behavior for all components using this component, its a global effect

File: packages/design-system/src/components/ai-command-bar/command-bar-button.tsx (hunk 2 above)
Comment (istarkov) [reply packages/design-system/src/components/ai-command-bar/command-bar-button.tsx:16]: We already wrapped it, for styling, just to make it look better

File: packages/design-system/src/components/ai-command-bar/command-bar.tsx (hunk 4 above)
Comment (istarkov) [reply packages/design-system/src/components/ai-command-bar/command-bar.tsx:66]: As I wrote I know :) Just to start review faster until I find better solution

Comment (kof) [reply packages/design-system/src/components/ai-command-bar/command-bar.tsx:66]: in such cases use todo statements so we don't assume this is how you wanted it to be

**Review by kof — APPROVED** [review]
please don't merge defaultProps like this

File: packages/design-system/src/components/button.tsx (hunk 1 above)
Comment (taylornowotny) [reply packages/design-system/src/components/button.tsx:54]: ok i'll add a dash. For figma it doesn't matter.

Comment (taylornowotny) [reply packages/design-system/src/components/button.tsx:54]: Screenshot 2023-10-07 at 9 58 42 AM
it is done

**Pull request merged**